packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -110,11 +110,13 @@ export function throwHttpGraphQLError<E extends Error>(
   );
 }
 
+const NODE_ENV = process.env.NODE_ENV;
+
 export async function runHttpQuery(
   handlerArguments: Array<any>,
   request: HttpQueryRequest,
```

Comment:
At this point in time, I'd expect `request.options.nodeEnv` to always be available, but the problem is that a call to `resolveGraphqlOptions` is later made which I _think_ can technically set `nodeEnv`. I don't know why anyone would do this, but it's technically possible based on the typing.

That means...